extensions/wikia/Wall/WallBaseController.class.php

```diff
@@ -364,8 +369,8 @@ public function statusInfoBox() {
 				$this->response->setVal( 'statusInfo', $info );
 				$this->response->setVal( 'id', $wallMessage->getId() );
 				if ( $showRemoveOrDeleteInfo ) {
-					$this->response->setVal( 'canRestore', $wallMessage->canRestore( $this->app->wg->User, false ) );
-					$this->response->setVal( 'fastrestore', $wallMessage->canFastRestore( $this->app->wg->User, false ) );
+					$this->response->setVal( 'canRestore', $wallMessage->canRestore( $this->app->wg->User ) );
```

Comment:
We could use the `User` object provided by the available `IContextSource`, as in https://camo.githubusercontent.com/8de516752b8503f0ff8859fa08d8cca80cc9d1c7/687474703a2f2f66617368696f6e61626c796765656b2e636f6d2f77702d636f6e74656e742f75706c6f6164732f323031342f31312f6b697474792d6361742d76696b696e672d6861742d30342e6a7067 . This way we make our code more testable as well by getting rid of global state.